### PR TITLE
Updated CMEMS JNB

### PR DESCRIPTION
**Pull request by TejasMorbagal**
[Description of PR]
Updated CMEMS JNB with the new dataset_id

Checklist:

* [ ] <strike>Add unit tests and/or doctests in docstrings</strike>
* [ ] <strike>Add docstrings and API docs for any new/modified user-facing classes and functions</strike>
* [ ] <strike>New/modified features documented in `docs/source/*`</strike>
* [x] Changes documented in `CHANGES.md`
* [x] AppVeyor CI passes
* [ ] <strike>Test coverage remains or increases (target 100%)</strike>

### CODE REVIEW

**Review by pont-us — CHANGES_REQUESTED** [review]
Looks mostly good and runs, but I'm a little confused about the handling of CMEMS access credentials:

1. The notebook uses the environment variable `CMEMS_USER` for the user name, whereas the plugin itself reads the user name from the environment variable `CMEMS_USERNAME`.
2. The notebook expects the credentials to be set in environment variables, but then reads them out using `os.getenv` and passes them directly as arguments to `new_data_store`. Why not skip this step and let the plugin read the credentials directly from environment variables? Or if there's a good reason to do it the current way, please add some text to explain why :).

**Comment by TejasMorbagal** [comment]
> Looks mostly good and runs, but I'm a little confused about the handling of CMEMS access credentials:
> 
>     1. The notebook uses the environment variable `CMEMS_USER` for the user name, whereas the plugin itself reads the user name from the environment variable `CMEMS_USERNAME`.
> 
>     2. The notebook expects the credentials to be set in environment variables, but then reads them out using `os.getenv` and passes them directly as arguments to `new_data_store`. Why not skip this step and let the plugin read the credentials directly from environment variables? Or if there's a good reason to do it the current way, please add some text to explain why :).

Yes, you are right. Both the points are resolved now.

**Review by pont-us — APPROVED** [review]
Looks good! Unfortunately the `open_data` step now fails, because (as far as I can tell) CMEMS has suddenly started returning some duplicate time-steps for `cmems_mod_glo_wav_anfc_0.083deg_PT3H-i`, which breaks the `sel` operation when trying to extract the time slice. But this is unconnected with the PR and I don't see any other problems, so approving :).

**Review by forman — CHANGES_REQUESTED** [review]
Please leave a note in `CHANGES.md`:

```markdown
* Updated example [Notebook for CMEMS data store](...) to reflect changes
  of parameter names that provide CMEMS API credentials.
```

**Comment by forman** [comment]
Also strike checks in PR description that do not apply rather than leaving them unticked.

File: examples/notebooks/datastores/7_cmems_data_store.ipynb [hunk 1]
```diff
@@ -104,16 +110,16 @@
...
         },
-        "cmems_user": {
-         "description": "Preferably set by environment variable CMEMS_USER",
-         "title": "CMEMS User",
-         "type": "string"
-        },
-        "cmems_user_password": {
+        "cmems_password": {
          "description": "Preferably set by environment variable CMEMS_PASSWORD",
          "title": "CMEMS User Password",
```
Comment (TonioF) [inline examples/notebooks/datastores/7_cmems_data_store.ipynb:115]:
```suggestion
         "title": "CMEMS Password",
```

**Review by TonioF — CHANGES_REQUESTED** [review]
Setting the cmems username and password as environment variables did not work for me (same procedure works fine in sh notebook). Passing them as store params works, though.

**Comment by TejasMorbagal** [comment]
> Please leave a note in `CHANGES.md`:
> 
> ```gfm
> * Updated example [Notebook for CMEMS data store](...) to reflect changes
>   of parameter names that provide CMEMS API credentials.
> ```

Done

File: examples/notebooks/datastores/7_cmems_data_store.ipynb (hunk 1 above)
Comment (TejasMorbagal) [reply examples/notebooks/datastores/7_cmems_data_store.ipynb:115]: I checked this again, it works fine for me. Note that environment variables are renamed to CMEMS_USERNAME and CMEMS_PASSWORD.

File: CHANGES.md [hunk 2]
```diff
@@ -2,6 +2,16 @@
...
 
+
+* Added Notebook 
+  [xcube-viewer-in-jl.ipynb](examples/notebooks/viewer/xcube-viewer-in-jl.ipynb)
+  that explains how xcube Viewer can now be utilised in JupyterLab
+  using the new (still experimental) xcube JupyterLab extension
+  [xcube-jl-ext](https://github.com/dcs4cop/xcube-jl-ext).
+  
+* Updated example [Notebook for CMEMS data store](examples/notebooks/datastores/7_cmems_data_store.ipynb) to reflect 
+changes of parameter names that provide CMEMS API credentials.
```
Comment (forman) [inline CHANGES.md:13]:
```suggestion
* Updated example 
  [Notebook for CMEMS data store](examples/notebooks/datastores/7_cmems_data_store.ipynb)
  to reflect changes of parameter names that provide CMEMS API credentials.
```

Comment (forman) [reply CHANGES.md:13]: Keep lines < 80 characters.

**Review by forman — CHANGES_REQUESTED** [review]

**Review by forman — APPROVED** [review]

**Review by AliceBalfanz — CHANGES_REQUESTED** [review]
Currently setting and using the env variables for cmems username and password only works if it is done before all the other xcube imports. So please move the cell and its description to the top of the notebook.

```
# os.environ["CMEMS_USERNAME"] =  ""
# os.environ["CMEMS_PASSWORD"] = ""
```

Furterhmore, in I get an error message when executing cell:
(Therefore, I was not able to test beyond this cell)
```
dataset = store.open_data('cmems_mod_glo_wav_anfc_0.083deg_PT3H-i', 
                          variable_names=['VHM0'], 
                          bbox=[9, 53, 20, 62], 
                          time_range=('2022-01-01','2022-01-05'))
dataset
```
Traceback:

```
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
Input In [11], in <cell line: 1>()
----> 1 dataset = store.open_data('cmems_mod_glo_wav_anfc_0.083deg_PT3H-i', 
      2                           variable_names=['VHM0'], 
      3                           bbox=[9, 53, 20, 62], 
      4                           time_range=('2022-01-01','2022-01-05'))
      5 dataset

File ~/miniconda3/envs/xcube/lib/python3.9/site-packages/xcube_cmems/store.py:350, in CmemsDataStore.open_data(self, data_id, opener_id, **open_params)
    348 def open_data(self, data_id: str, opener_id: str = None,
    349               **open_params) -> Any:
--> 350     return self._get_opener(opener_id=opener_id).open_data(data_id,
    351                                                            **open_params)

File ~/miniconda3/envs/xcube/lib/python3.9/site-packages/xcube_cmems/store.py:250, in CmemsDataOpener.open_data(self, data_id, **open_params)
    248 ds = self.open_dataset(data_id)
    249 if open_params:
--> 250     ds = self.subset_cube_with_open_params(ds, **open_params)
    251 return ds

File ~/miniconda3/envs/xcube/lib/python3.9/site-packages/xcube_cmems/store.py:214, in CmemsDataOpener.subset_cube_with_open_params(ds, **open_params)
    210 @staticmethod
    211 def subset_cube_with_open_params(ds: xr.Dataset,
    212                                  **open_params) -> xr.Dataset:
    213     if 'time_range' in open_params:
--> 214         ds = ds.sel(time=slice(open_params.get('time_range')[0],
    215                                open_params.get('time_range')[1]))
    216     if 'bbox' in open_params:
    217         x_y_var_name_pairs = [
    218             {'y': 'latitude', 'x': 'longitude'},
    219             {'y': 'lat', 'x': 'lon'},
    220             {'y': 'y', 'x': 'x'}
    221         ]

File ~/miniconda3/envs/xcube/lib/python3.9/site-packages/xarray/core/dataset.py:2565, in Dataset.sel(self, indexers, method, tolerance, drop, **indexers_kwargs)
   2504 """Returns a new dataset with each array indexed by tick labels
   2505 along the specified dimension(s).
   2506 
   (...)
   2562 DataArray.sel
   2563 """
   2564 indexers = either_dict_or_kwargs(indexers, indexers_kwargs, "sel")
-> 2565 query_results = map_index_queries(
   2566     self, indexers=indexers, method=method, tolerance=tolerance
   2567 )
   2569 if drop:
   2570     no_scalar_variables = {}

File ~/miniconda3/envs/xcube/lib/python3.9/site-packages/xarray/core/indexing.py:183, in map_index_queries(obj, indexers, method, tolerance, **indexers_kwargs)
    181         results.append(IndexSelResult(labels))
    182     else:
--> 183         results.append(index.sel(labels, **options))
    185 merged = merge_sel_results(results)
    187 # drop dimension coordinates found in dimension indexers
    188 # (also drop multi-index if any)
    189 # (.sel() already ensures alignment)

File ~/miniconda3/envs/xcube/lib/python3.9/site-packages/xarray/core/indexes.py:442, in PandasIndex.sel(self, labels, method, tolerance)
    439 coord_name, label = next(iter(labels.items()))
    441 if isinstance(label, slice):
--> 442     indexer = _query_slice(self.index, label, coord_name, method, tolerance)
    443 elif is_dict_like(label):
    444     raise ValueError(
    445         "cannot use a dict-like object for selection on "
    446         "a dimension that does not have a MultiIndex"
    447     )

File ~/miniconda3/envs/xcube/lib/python3.9/site-packages/xarray/core/indexes.py:215, in _query_slice(index, label, coord_name, method, tolerance)
    207 indexer = index.slice_indexer(
    208     _sanitize_slice_element(label.start),
    209     _sanitize_slice_element(label.stop),
    210     _sanitize_slice_element(label.step),
    211 )
    212 if not isinstance(indexer, slice):
    213     # unlike pandas, in xarray we never want to silently convert a
    214     # slice indexer into an array indexer
--> 215     raise KeyError(
    216         "cannot represent labeled-based slice indexer for coordinate "
    217         f"{coord_name!r} with a slice over integer positions; the index is "
    218         "unsorted or non-unique"
    219     )
    220 return indexer

KeyError: "cannot represent labeled-based slice indexer for coordinate 'time' with a slice over integer positions; the index is unsorted or non-unique"
```

**Comment by AliceBalfanz** [comment]
> This used to work before. It is a new issue. I will create a new issue in xcube_cmems for this and work on it

Please still adjust it in the Notebook as it should be working with the current version as well.

**Comment by TejasMorbagal** [comment]
> > This used to work before. It is a new issue. I will create a new issue in xcube_cmems for this and work on it
> 
> Please still adjust it in the Notebook as it should be working with the current version as well.

The issue comes from CMEMS Data provider as they have duplicate times returned for the Dataset ID used in the example notebook. I propose to use a different dataset ID which works in context of this PR.

**Comment by AliceBalfanz** [comment]
In the cell of setting the environment variables, you also need to add the import: 

```
import os

# os.environ["CMEMS_USERNAME"] =  ""
# os.environ["CMEMS_PASSWORD"] = ""
```

otherwise it wont work with the current version. 

Please also replace the last cell with: 

`dataset.sea_surface_temperature.isel(time=-1).plot(figsize=(10, 10))`

It will plot the last timestamp of the dataset, which is a nicer plot, than plotting the first one. Using `isel` is following the style of how we select time stamps in other notebooks.

**Pull request merged**